lib/compiler/src/beam_ssa_type.erl

```diff
@@ -885,20 +972,26 @@ infer_type({bif,element}, [#b_literal{val=Pos},#b_var{}=Tuple], _Ds) ->
         true ->
             []
     end;
-infer_type({bif,'=:='}, [#b_var{}=Src,#b_literal{}=Lit], Ds) ->
-    Def = maps:get(Src, Ds),
-    Type = get_type(Lit, #{}),
-    [{Src,Type}|infer_tuple_size(Def, Lit) ++
-         infer_first_element(Def, Lit)];
+infer_type({bif,element}, [#b_var{}=Position,#b_var{}=Tuple], _Ds) ->
+    [{Position,t_integer()},{Tuple,#t_tuple{}}];
```

Comment:
It is highly unlikely that there is any type information available for the position variable (except it being an integer). Since `infer_type/3` currently does not have a `Ts` argument (type database), I will not change this code.